### PR TITLE
Stage 2 – Development environment and CI/CD

### PR DESCRIPTION
**Pull request by YaroslavKSE**

### CODE REVIEW

File: react-frontend/package.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
   "description": "- react@18",
   "main": "index.js",
   "scripts": {
-    "build": "webpack --env dev",
-    "dev": "webpack serve --env dev",
+    "start": "webpack serve --config ./webpack.config.js --mode development",
```
Comment (denis-zavgorodny) [inline react-frontend/package.json]: If you want to use `--mode` key, you need to update your `webpack` configuration and use something like
```js
module.exports = (env, argv) => {
  return {
    mode: argv.mode
...
```

And please, check other places where we rely on `env.dev` parameter

File: react-frontend/Dockerfile-frontend [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+COPY package*.json ./
+
+# Install dependencies
+RUN npm install
+
+# Copy the rest of the application files to the working directory
+COPY . .
+
+# Build the React application for production
+RUN npm run build
```
Comment (denis-zavgorodny) [inline react-frontend/Dockerfile-frontend:16]: We build our project outside of docker, can we just copy our artifacts there? We do not need node modules and even nodejs here.

Comment (denis-zavgorodny) [reply react-frontend/Dockerfile-frontend:16]: And then just build the image in github workflow

Comment (YaroslavKSE) [reply react-frontend/Dockerfile-frontend:16]: I am going to use docker hub for deployment and versioning, so I need seperate build process which is going to be triggered on every push to main branch. On pr I would use your CI to satisfy the requirements of the stage 2 and maybe in future I am going to add tests on backend and frontend. Moreover, I use docker while developing so I can't copy artifacts from github action.

File: react-frontend/package.json (hunk 1 above)
Comment (YaroslavKSE) [reply react-frontend/package.json]: Okay, I will research this topic and make changes

File: react-frontend/Dockerfile-frontend (hunk 2 above)
Comment (denis-zavgorodny) [reply react-frontend/Dockerfile-frontend:16]: 👍 makes sense

File: react-frontend/package.json (hunk 1 above)
Comment (YaroslavKSE) [reply react-frontend/package.json]: Refactored in [459af7c](https://github.com/YaroslavKSE/AuctionWeb/pull/1/commits/459af7cffdf95c302983c2c9a20273574b610bbf)

**Review by denis-zavgorodny — APPROVED** [review]

**Pull request merged**